args/audio/data/importers/audio_importers.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+#include <core/core.hpp>
+#include <audio/data/audio_segment.hpp>
+
+namespace args::audio
+{
+    struct mp3_audio_loader : public fs::resource_converter<audio_segment, audio_import_settings>
+    {
+        virtual common::result_decay_more<audio_segment, fs_error> load(const fs::basic_resource& resource, audio_import_settings&& settings) override;
+        static std::shared_ptr<ALCcontext> context;
+    private:
+        static async::readonly_rw_spinlock contextLock;
```

Comment:
this is not the same lock as the one in audiosystem.hpp, audiosystem and the audio loader can thus both lock different locks at the same time. meaning they can both try to use the context at the same time.